core/src/main/java/hudson/model/Job.java

```diff
@@ -679,7 +670,54 @@ public void movedTo(DirectlyModifiableTopLevelItemGroup destination, AbstractIte
         }
     }
 
-    @Override public void delete() throws IOException, InterruptedException {
+    @Override
+    public void delete() throws IOException, InterruptedException {
+        checkPermission(DELETE);
+        // if a build is in progress. Cancel it.
+        if (this instanceof Queue.Task) {
+            // clear any items in the queue so they do not get picket up
+            Queue.getInstance().cancel((Queue.Task) this);
+            // interrupt any builds in progress
+            Map<Queue.Executable, Executor> building = new LinkedHashMap<>();
+            for (Computer c : Jenkins.getInstance().getComputers()) {
+                for (Executor e : c.getOneOffExecutors()) {
+                    WorkUnit workUnit = e.getCurrentWorkUnit();
+                    if (workUnit != null && (workUnit.work == this || workUnit.work.getOwnerTask() == this)) {
+                        building.put(e.getCurrentExecutable(), e);
+                        e.interrupt();
+                    }
+                }
+                for (Executor e : c.getExecutors()) {
+                    WorkUnit workUnit = e.getCurrentWorkUnit();
+                    if (workUnit != null && (workUnit.work == this || workUnit.work.getOwnerTask() == this)) {
+                        building.put(e.getCurrentExecutable(), e);
+                        e.interrupt();
+                    }
+                }
+            }
+            if (!building.isEmpty()) {
+                // give them 15 seconds or so to respond to the interrupt
+                long expiration = System.nanoTime() + TimeUnit.SECONDS.toNanos(15);
+                // comparison with executor.getCurrentExecutable() == computation currently should always be true
+                // as we no longer recycle Executors, but safer to future-proof in case we ever revisit recycling
+                while (!building.isEmpty() && expiration - System.nanoTime() > 0L) {
+                    for (Iterator<Map.Entry<Queue.Executable, Executor>> iterator = building.entrySet().iterator();
+                         iterator.hasNext(); ) {
+                        Map.Entry<Queue.Executable, Executor> entry = iterator.next();
+                        // comparison with executor.getCurrentExecutable() == executable currently should always be true
+                        // as we no longer recycle Executors, but safer to future-proof in case we ever revisit recycling
+
+                        if (!entry.getValue().isAlive() || entry.getKey() != entry.getValue().getCurrentExecutable()) {
+                            iterator.remove();
+                        }
+                    }
+                    Thread.sleep(50L);
+                }
+                if (!building.isEmpty()) {
+                    throw new AbortException(Messages.Job_FailureToStopBuilds(building.size(), getFullDisplayName()));
```

Comment:
I do not see why. The user has gotten a failure message (I hope—test it?); the job has not been deleted; there is nothing for the admin to do.